packages/react-router/modules/Router.js

```diff
@@ -46,13 +57,9 @@ class Router extends React.Component {
     };
   }
 
-  componentWillMount() {
-    const { children, history } = this.props;
-
-    invariant(
-      children == null || React.Children.count(children) === 1,
-      "A <Router> may have only one child element"
-    );
+  constructor(props) {
```

Comment:
The subscription can't go in the constructor. If the component is removed from the tree before rendering (which may happen in async rendering), we'll leak a subscription to the history instance. The subscription needs to be put in a componentDidMount.